spec.bs

```diff
@@ -676,3 +683,190 @@ To <dfn>determine the numeric value</dfn> of a [=signal base value=]
 {{double}}.
 
 Issue: Fill in.
+
+Privacy considerations {#privacy-considerations}
+================================================
+
+<em>This section is non-normative.</em>
+
+Cross-site information disclosure {#cross-site-information-disclosure}
+----------------------------------------------------------------------
+
+This API lets isolated contexts with access to cross-site data (i.e. <a
+href="https://wicg.github.io/shared-storage/">Shared Storage</a> worklets/<a
+href="https://wicg.github.io/turtledove/">Protected Audience</a> script runners)
+send aggregatable reports over the network.
+
+Aggregatable reports contain encrypted high entropy cross-site information, in
+the form of key-value pairs (i.e. contributions to a histogram). The information
+embedded in the contributions is arbitrary but can include things like browsing
+history and other cross-site activity. The API aims to protect this information
+from being passed from one site to another.
+
+### Restricted contribution processing ### {#restricted-contribution-processing}
+
+The histogram contributions are not exposed directly. Instead, they are
+encrypted so that they can only be processed by a trusted aggregation service.
+This trusted aggregation service sums the values across the reports for each key
+and adds noise to each of these values to produce ‘summary reports’.
+
+The output of that processing will be an aggregated, noised histogram. The
+service ensures that any report can not be processed multiple times. Further,
+information exposure is limited by contribution budgets on the client. In
+principle, this framework can support specifying a noise parameter which
+satisfies differential privacy.
+
+### Unencrypted metadata ### {#unencrypted-metadata}
+
+These reports also expose a limited amount of metadata, which is not based on
+cross-site data. The recipient of the report may also be able to observe
+side-channel information such as the time when the report was sent, or IP
+address of the sender.
+
+### Protecting against leaks via the number of reports ### {#protecting-against-leaks-via-the-number-of-reports}
+
+However, the number of reports with the given metadata could expose some
+cross-site information. To protect against this, the API delays sending reports
+by a randomized amount of time to make it difficult to determine whether a
+report was sent or not from any particular event. In the case that a context ID
```

Comment:
Could we add an issue to spec the null report generation with context ID?

Reply:
Updated an issue to be clearer